src/features/calendar/utils/clusterEventsForWeekCalender.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { CLUSTER_TYPE } from 'features/campaigns/hooks/useClusteredActivities';
+import clusterEventsForWeekCalender from './clusterEventsForWeekCalender';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import {
+  ACTIVITIES,
+  EventActivity,
+} from 'features/campaigns/models/CampaignActivitiesModel';
+
+const mockEventData: ZetkinEvent = {
+  activity: {
+    id: 1,
+    title: 'Flyering',
+  },
+  campaign: {
+    id: 1,
+    title: 'My campaign',
+  },
+  cancelled: null,
+  contact: null,
+  end_time: '1857-07-05T13:37:00.000Z',
+  id: 1,
+  info_text: '',
+  location: {
+    id: 1,
+    lat: 0,
+    lng: 0,
+    title: 'Dorfplatz',
+  },
+  num_participants_available: 0,
+  num_participants_required: 0,
+  organization: {
+    id: 1,
+    title: 'KPD',
+  },
+  published: null,
+  start_time: '1857-07-05T13:37:00.000Z',
+};
+
+function mockEvent(id: number, data: Partial<ZetkinEvent>): EventActivity {
+  return {
+    data: {
+      ...mockEventData,
+      ...data,
+      id,
+    },
+    endDate: null,
+    kind: ACTIVITIES.EVENT,
+    startDate: null,
+  };
+}
```

Comment:
Maybe whack this in the test utils?

Reply:
I didn't because it's a very specialized test utility for this test, because it only allows you to create event activities (not events) and only using certain conditions (like no events are published).